src/database/controllers/article.controller.ts

```diff
@@ -0,0 +1,20 @@
+import { Controller, Get, Param, Query } from '@nestjs/common';
+import { ArticleService } from '../article.service';
+import { Article } from '../schemas/article.schema';
+import { Types } from 'mongoose';
+
+@Controller('article')
+export class ArticleController {
+  constructor(private readonly articleService: ArticleService) {}
+
+  @Get()
+  async findAll(@Query() query: any): Promise<Article[]> {
+    const { filter, ...options } = query;
+    return this.articleService.findAll(filter, options);
+  }
```

Comment:
Please avoid using 'any'. Instead, create a query type that includes 'q' and 'limit'. Afterward, we performed some post-processing on the data. For reference, please see [v1](https://github.com/NCTU-SYNC/sync-server/blob/76ccd329f33a12b1b5b333e51b86046fcc9fb8ec/src/controllers/article_controller.js#L156).